pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -451,15 +451,27 @@ func monitorFor(p *Propagator, clientPool dynamic.ClientPool, resource unversion
 				// namespaces if it's namespace scoped, so leave
 				// APIResource.Namespaced as false is all right.
 				apiResource := unversioned.APIResource{Name: resource.Resource}
-				return client.Resource(&apiResource, api.NamespaceAll).List(&options)
+				// The parameter codec used by the dynamic client cannot encode
+				// api.ListOptions.
+				versionedOptions := &v1.ListOptions{}
+				if err := api.Scheme.Convert(&options, versionedOptions); err != nil {
+					return nil, err
+				}
+				return client.Resource(&apiResource, api.NamespaceAll).List(versionedOptions)
 			},
 			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
 				// APIResource.Kind is not used by the dynamic client, so
 				// leave it empty. We want to list this resource in all
 				// namespaces if it's namespace scoped, so leave
 				// APIResource.Namespaced as false is all right.
 				apiResource := unversioned.APIResource{Name: resource.Resource}
-				return client.Resource(&apiResource, api.NamespaceAll).Watch(&options)
+				// The parameter codec used by the dynamic client cannot encode
+				// api.ListOptions.
+				versionedOptions := &v1.ListOptions{}
+				if err := api.Scheme.Convert(&options, versionedOptions); err != nil {
```

Comment:
If you're going to call convert here, then I agree that it's better to solve the problem by adding a parameter codec that does this for you.

My recommendation is to not do this convert here, and instead make all callers of this function use v1.ListOptions, etc.